### ISSUE TITLE
STU 3.0.1 Appointment test uses invalid resource and problem with conditional interactions

### ISSUE THREAD

**Issue by RamSoftDev** [fhir-crucible/crucible#282]
* X010: Appointment: Create New: Failed Fhir Schema Validation because Reason element in Appointment resource (https://www.hl7.org/fhir/appointment.html#resource) is supposed to be an array of codeablConcept, but data used in X010 test has reason as a single codeableConcept which causes all subsequent tests to fail.  

* Even though server declares that it supports conditional interactions such as conditional create, the corresponding test cases won't be included.

**Comment by arscan** [comment]
For the X010, you are correct.  I have a PR open for a fix here: https://github.com/fhir-crucible/plan_executor/pull/142

For the conditional interactions getting filtered out -- we cache the conformance statement.  There is a refresh button in the Capability Statement tab, clicking it appears to have updated our cached copy.  Could you verify that it is now not filtered out?

Thanks for the report!

**Comment by RamSoftDev** [comment]
Thanks for your quick response.

I refreshed the Capability statement, and it's the same.
Maybe i interpreted incorrectly, what i mean is that in server's Capability Statement, it shows everything correctly, Conditional Update : true and so on. However, when *supported only* filter is added, all the test cases related conditional create/update X012, X013, X014, X032, X033 and X034 are filtered out.

**Comment by arscan** [comment]
You are correct, thanks.  It doesn't appear that we are properly filtering conditionalUpdate, conditionalRead, conditionalDelete here:

https://github.com/fhir-crucible/crucible/blob/master/app/models/server.rb#L198

We handle operations and interactions, but not the conditionals, which are represented differently in the Capability Statement.

**Comment by Rende11** [comment]
Base Resources/Resource Test Appointment/Appointment: Create New

In request payload, we have "reason" field which has object value - "coding". Isn't "coding" should be into an array?
